### PR TITLE
Updated logic for parsing date of last contribution from cookie

### PR DESCRIPTION
**Pull request by Mullefa**
## What does this change?

The last contribution timestamp can have 2 different formats - ISO 8601 (contributions-frontend) and Epoch time in milliseconds (support-frontend). The updated logic accounts for that.

I've also updated the method name `isRecentContributor` to `isRecentOneOffContributor` since we now accept recurring contributions as well.

### CODE REVIEW

**Review by jranks123 — APPROVED** [review]
Looks good, save for the failing tests

File: static/src/javascripts/projects/common/modules/commercial/user-features.js [hunk 1]
```diff
@@ -147,14 +150,64 @@ const isPayingMember = (): boolean =>
...
-const daysSinceLastContribution = daysSince(lastContributionDate);
+    // Contributions frontend set date time of last contribution using ISO 8601 format.
+    // Support frontend set date time of last contribution in Epoch milliseconds.
+    // If we first attempted to parse cookie in Epoch milliseconds format (parseInt()),
+    // then a value in ISO 8601 format would parse (incorrectly) e.g. 2018-08-17T16:11:10Z => 2018
+    // So first attempt to parse cookie in ISO 8601 format.
+
+    const parseIso = (c: string) => {
+        try {
+            const ms = Date.parse(c);
```
Comment (joelochlann) [inline static/src/javascripts/projects/common/modules/commercial/user-features.js]: Definitely a good call to be defensive, but `Date.parse()` and `parseInt()` don't throw exceptions (unlike `JSON.parse()`!) so it's OK to ditch the try/catch

**Review by joelochlann — APPROVED** [review]
Just one tiny comment.

I wonder if it could be worth adding some logic in a separate PR to replace cookies in non-epoch time with epoch time. Specifically, for people who had their cookie value generated by `new Date().toString()` (e.g. `"Mon Aug 20 2018 16:46:54 GMT+0100 (British Summer Time)"
` which breaks cookie parsing on identity. That would ensure identity works for them.

**Comment by Mullefa** [comment]
@joelochlann - good shout

**Comment by jfsoul** [comment]
@joelochlann I did a bit more digging on that cookie parsing issue.  It's specifically related to our service for social sign-in which is using an old framework version (spray).  If we were to update the framework, the parsing has been made a bit more lenient, so this issue would not occur.

Doesn't hurt to have consistent and (fully) valid cookie values though!

**Pull request merged**